src/probnum/random_variables/_normal.py

```diff
@@ -0,0 +1,582 @@
+            scipy.stats.norm.entropy(loc=self._mean, scale=self._cov),
+            dtype=np.float_,
+        )
+
+    # Multi- and matrixvariate Gaussians
+    def _dense_cov_cholesky(self) -> np.ndarray:
+        eps = 10 ** -12  # damping needed to avoid negative definite covariances
+
+        return scipy.linalg.cholesky(
+            self._dense_cov + eps * np.eye(self._dense_cov.shape[0], dtype=self.dtype),
+            lower=True,
+        )
+
+    def _dense_sample(self, size: ShapeType = ()) -> np.ndarray:
+        sample = scipy.stats.multivariate_normal.rvs(
+            mean=self._dense_mean.ravel(),
+            cov=self._dense_cov,
+            size=size,
+            random_state=self.random_state,
+        )
+
+        return sample.reshape(sample.shape[:-1] + self.shape)
+
+    @staticmethod
+    def _arg_todense(x: Union[np.ndarray, linops.LinearOperator]) -> np.ndarray:
+        if isinstance(x, linops.LinearOperator):
+            return x.todense()
+        elif isinstance(x, np.ndarray):
+            return x
+        else:
+            raise ValueError(f"Unsupported argument type {type(x)}")
+
+    @staticmethod
+    def _dense_in_support(x: _ValueType) -> bool:
+        return np.all(np.isfinite(Normal._arg_todense(x)))
+
+    def _dense_pdf(self, x: _ValueType) -> np.float_:
+        return scipy.stats.multivariate_normal.pdf(
+            Normal._arg_todense(x).reshape(x.shape[: -self.ndim] + (-1,)),
+            mean=self._dense_mean.ravel(),
+            cov=self._dense_cov,
+        )
+
+    def _dense_logpdf(self, x: _ValueType) -> np.float_:
+        return scipy.stats.multivariate_normal.logpdf(
+            Normal._arg_todense(x).reshape(x.shape[: -self.ndim] + (-1,)),
+            mean=self._dense_mean.ravel(),
+            cov=self._dense_cov,
+        )
+
+    def _dense_cdf(self, x: _ValueType) -> np.float_:
+        return scipy.stats.multivariate_normal.cdf(
+            Normal._arg_todense(x).reshape(x.shape[: -self.ndim] + (-1,)),
+            mean=self._dense_mean.ravel(),
+            cov=self._dense_cov,
+        )
+
+    def _dense_logcdf(self, x: _ValueType) -> np.float_:
+        return scipy.stats.multivariate_normal.logcdf(
+            Normal._arg_todense(x).reshape(x.shape[: -self.ndim] + (-1,)),
+            mean=self._dense_mean.ravel(),
+            cov=self._dense_cov,
+        )
+
+    def _dense_entropy(self) -> np.float_:
+        return _utils.as_numpy_scalar(
+            scipy.stats.multivariate_normal.entropy(
+                mean=self._dense_mean.ravel(),
+                cov=self._dense_cov,
+            ),
+            dtype=np.float_,
+        )
+
+    # Matrixvariate Gaussian with Kronecker covariance
+    def _kronecker_cov_cholesky(self) -> linops.Kronecker:
+        assert isinstance(self._cov, linops.Kronecker)
+
+        A = self._cov.A.todense()
+        B = self._cov.B.todense()
+
+        eps = 10 ** -12  # damping needed to avoid negative definite covariances
```

Comment:
ideally this should be a global constant and longterm maybe even adjusted based on the precision used. For now at least a constant across normal distributions would be good.
```python
COV_CHOLESKY_DAMPING = 10 **-12
```